Main.py

```diff
@@ -23,33 +23,43 @@ def App():
         print ("Please enter the index of the news source or press 99 to quit")
         try:
             newsSourceNumber=raw_input("News Source Number >>>> ")
+            newsSourceNumber=int(newsSourceNumber)
         except ValueError:  
             print ("That is not a valid News Source Number")
-        newsSourceNumber=int(newsSourceNumber)
         if newsSourceNumber==99:
             sys.exit()
         if (newsSourceNumber >=len(newsSources)):
             print ("Please select the index no less than "+ str(len(newsSources)))  
         obj=NewsPulling(newsSources[newsSourceNumber])
-        Articles=obj.BeautifyArticles();   
+        Articles=obj.BeautifyArticles(); 
+        extr = None
+        pastarticleNum = -1
         while True:
             print ("Do you want to read any story further? If yes, please select the number corresponding to the article")
             print ("Press 66 to go back to the main menu")
             print ("Press 99 to quit")
+            if extr != None:
+                print ("Press s to save the article as a file")
             try:
                 articleNumber=raw_input("Article No >>>> ")
+                articleNumber=int(articleNumber)
             except ValueError:
-                print ("That is not a valid Article Number")
-            articleNumber=int(articleNumber)
-            if articleNumber==99 :
+                if articleNumber == 's' and extr != None:

```

Comment:
since you have `articleNumber=int(articleNumber)`  pressing s would lead to value error and this line will never be executed.